central/namespace/datastore/datastore.go

```diff
@@ -189,6 +191,14 @@ func (b *datastoreImpl) GetNamespaces(ctx context.Context) ([]*storage.Namespace
 	return allowedNamespaces, nil
 }
 
+func (b *datastoreImpl) GetBatch(ctx context.Context, ids []string) ([]*storage.NamespaceMetadata, error) {
+	namespaces, _, err := b.store.GetMany(ctx, ids)
```

Comment:
Check for permission first.
```
var namespaces []*storage.NamespaceMetadata
var err error
if ok, err := namespaceSAC.ReadAllowed(ctx); err != nil {
		return nil, err
} else if !ok {
		return ds.SearchNamespaces(ctx, {namespace_query})
}
namespaces, _, err = b.store.GetMany(ctx, ids)
...
```

Reply:
Further explained: The above block checks for global access to `namespace` SAC resource. If not, delegate the filtering based on permissions to the searcher.